platforms/nuttx/NuttX/Make.defs.in

```diff
@@ -33,19 +33,45 @@
 #
 ############################################################################
 
+# Set flags for the NuttX build system for Cygwin
+# They are already used by Config.mk
+ifneq (, $(findstring CYGWIN, $(shell uname)))
+  CONFIG_HOST_CYGWIN := y
+  CONFIG_WINDOWS_CYGWIN := y
+  CONFIG_CYGWIN_WINTOOL := y
+endif
+
 include $(TOPDIR)/.config
 include $(TOPDIR)/tools/Config.mk
 
+# Replace each space separated word in the string that contains "cygdrive"
+# with the windows path and escaped backslashes
+# e.g. [hello -I /cygdrive/c -I/cygdrive/c] -> [hello -I C:\\ ]
+define cygwin_to_windows_paths
+$(subst \,\\,$(foreach word,$(1),$(shell \
+if [[ "$(word)" == *"cygdrive"* ]]; then \
+  cygpath -w "$(word)"; \
+else \
+  echo $(word); \
+fi \
+)))
+endef
+
 CINCPATH := $(shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include)
 CXXINCPATH := $(shell $(INCDIR) -s "$(CC)" $(TOPDIR)$(DELIM)include$(DELIM)cxx)
 
 ARCHINCLUDES += $(CINCPATH)
 ARCHXXINCLUDES += $(CINCPATH) $(CXXINCPATH)
+ARCHSCRIPT = -T$(BOARD_DIR)$(DELIM)scripts$(DELIM)flash.ld
 
-ifeq ($(CONFIG_CYGWIN_WINTOOL),y)
-  ARCHSCRIPT = -T "$(shell cygpath -w $(BOARD_DIR)$(DELIM)scripts$(DELIM)flash.ld)"
-else
-  ARCHSCRIPT = -T$(BOARD_DIR)$(DELIM)scripts$(DELIM)flash.ld
+ifeq ($(CONFIG_BOARD_USE_PROBES),y)
+   ARCHINCLUDES += -I $(TOPDIR)/arch/$(CONFIG_ARCH)/src/$(CONFIG_ARCH_CHIP) -I $(TOPDIR)/arch/$(CONFIG_ARCH)/src/common
```

Comment:
@davids5 I shifted these additions up because they need to get converted as well and added spaces between `-I` and the path for simpler makefile parsing.